lib/git_reflow.rb

```diff
@@ -107,42 +108,34 @@ def deliver(options = {})
     feature_branch    = current_branch
     base_branch       = options['base'] || 'master'
 
-    update_current_branch
-    fetch_destination(base_branch)
-    update_destination(feature_branch)
-
     begin
       existing_pull_request = git_server.find_open_pull_request( :from => current_branch, :to => base_branch )
 
       if existing_pull_request.nil?
         say "No pull request exists for #{remote_user}:#{current_branch}\nPlease submit your branch for review first with \`git reflow review\`", :deliver_halted
       else
 
-        commit_message = if "#{existing_pull_request.description}".length > 0
-                           existing_pull_request.description
-                         else
-                           "#{get_first_commit_message}"
-                         end
-
         if existing_pull_request.good_to_merge?(force: options['skip_lgtm'])
           puts "Merging pull request ##{existing_pull_request.number}: '#{existing_pull_request.title}', from '#{existing_pull_request.feature_branch_name}' into '#{existing_pull_request.base_branch_name}'"
 
-          update_destination(base_branch)
-          merge_feature_branch(feature_branch,
-                               :destination_branch  => base_branch,
-                               :pull_request_number => existing_pull_request.number,
-                               :lgtm_authors        => existing_pull_request.approvals,
-                               :message             => commit_message)
-          committed = run_command_with_label 'git commit', with_system: true
-
-          if committed
-            say "Merge complete!", :success
+          begin
+            res = existing_pull_request.merge!(
+              feature_branch,
```

Comment:
we also don't need the `feature_branch` here as we can get this from within our `merge!` method with `GitReflow.current_branch`